ir/src/trace_columns.rs

```diff
@@ -1,39 +1,36 @@
-use super::{BTreeMap, SemanticError};
+use super::TraceSegment;
 
-/// A representation of the trace columns declared for the AIR.
-#[derive(Debug, Default)]
+/// Describes columns in the execution trace by the trace segment to which it belongs, it's size
+/// and it's offset.
+#[derive(Debug, Copy, Clone, Eq, PartialEq)]
 pub struct TraceColumns {
-    /// A map of a set of trace columns using the declared identifier as the key and the column
-    /// index as the value.
-    columns: BTreeMap<String, usize>,
+    trace_segment: TraceSegment,
+    size: usize,
+    offset: usize,
```

Comment:
Makes sense. My thinking was that the size is a more local property of the column group and should come before the offset which is meaningful in the context of the whole trace. But this argument probably doesn't make sense since `trace_segment` is also in the context of the whole trace. Will change it.